src/components/item/form/AppForm.tsx

```diff
@@ -168,39 +196,71 @@ const AppForm = ({ onChange, updatedProperties = {} }: Props): JSX.Element => {
     );
   }
   return (
-    <Stack direction="column" height="100%" spacing={2} minHeight="0px">
-      <TextField
-        fullWidth
-        placeholder={translateBuilder('Search for an app')}
-        variant="outlined"
-        autoFocus
-        size="small"
-        onChange={searchAnApp}
-      />
-      <Box display="flex" flexGrow={1} minHeight="0px" overflow="scroll" p={1}>
-        <Grid2
-          container
-          spacing={2}
-          height="max-content"
-          maxHeight={400}
-          alignItems="stretch"
+    <Stack spacing={2} mt={0.5}>
+      <Stack
+        direction="column"
+        height="100%"
+        spacing={1}
+        bgcolor={alpha('#ADADAD', 0.06)}
+        padding={2}
+        borderRadius={2}
+      >
+        <TextField
+          fullWidth
+          placeholder={translateBuilder(BUILDER.ADD_BUILT_IN_APP_TEXT)}
+          variant="outlined"
+          autoFocus
+          size="small"
+          onChange={searchAnApp}
+        />
+        <Stack direction="row" alignItems="center" spacing={1}>
+          <Typography variant="label">
+            {hasMostUsedData && !showFull
+              ? translateBuilder(BUILDER.APP_SECTION_MOST_USED)
+              : translateBuilder(BUILDER.APP_SECTION_ALL_APP)}
+          </Typography>
+          <Button
+            size="small"
+            variant="text"
+            onClick={() => setShowFull(!showFull)}
+            id={CUSTOM_APP_CYPRESS_ID}
+          >
+            {hasMostUsedData &&
+              (!showFull
+                ? translateBuilder(BUILDER.APP_SECTION_ALL_APP_BUTTON_TEXT)
+                : translateBuilder(BUILDER.APP_SECTION_MOST_USED_BUTTON_TEXT))}
+          </Button>
+        </Stack>
+        <Stack
+          flexDirection="row"
+          flexWrap="wrap"
+          width="100%"
+          gap={1}
+          height="300px"
```

Comment:
You should try to get rid of this random value. Probably a working idea is to include to rest of the form in the `Stack` as well. You can also set some `minHeight` or `maxHeight`.